### ISSUE TITLE
Nested LESS imports don't respect current file's directory

### ISSUE THREAD

**Issue by rblackman** [less/less.js#2321]
When importing a less file, if that file also imports a less file, the nested import will try to import the file relative to the initial file's path.

This only happens when using the ```@import url('foo')``` format it works if you use the ```@import 'foo'``` format.

So for example, if you have the following directory structure:

```
- project/
  | - less/
      | - _global.less
      | - _import.less
      | - modules/
          | - button.less
```

_import.less:
```less
@import url('./_global.less');
...
```
button.less:
```less
@import url('../import.less');
...
```

```_import.less``` will look for ```_global.less``` in the ```project/``` directory.

**Comment by lukeapage** [comment]
I have a feeling this has always happened but I am not sure if it is documented or intended behaviour

**Comment by matthew-dean** [comment]
AFAIK, imports should always be file relative, since they're treated as an extension of CSS imports. _import.less is establishing an import dependency that should resolve the same no matter what other file imports _import.less.

Now, with this line ```@import url('./_global.less');``` in _import.less, if, as you say, it's looking for _global.less in ```project/``` and not ```/project/less``` (since _import.less is in ```/project/less``` then that's not right.

**Comment by lukeapage** [comment]
I agree, im jyst saying i gave vague memories of a discussion around people
wanting imports absolute using it with a url(). But it was a long long time
ago.

I would just double check our documentation agrees its a bug before fixing.

**Comment by matthew-dean** [comment]
Hmm, I had a look at the ```@import``` documentation, and it doesn't really explain that imports are file-relative, as far as I could tell. It should probably be added to the docs.

I also recall where people wanted to dynamically import a "local" version of an import for stuff like theming. That is, they wanted to set a starting theme directory, and then have all those import directives refer to the local theme's files (Issue...?). However, I think that could be accomplished by addressing variable interpolation within nested imports (Issue... ?), although I know you've said it's a complex problem of the order of @import / variable evaluation. Regardless, I don't think a behavioural change is needed. Unless @rblackman says otherwise, it appears to be expected behavior as you said, but could use some supporting documentation.

**Comment by SomMeri** [comment]
I read http://lesscss.org/features/#import-directives-feature and it does not says whether the import should be relative or absolute. 

That being said, I failed to reproduce the problem. I recreated the above directory structure and tried to compile button.less from three different directories:
* `lessc project\modules\button.less`
* `lessc modules\button.less`
* `lessc button.less`

All three attempts interpreted imports relatively (i added ruleset with filename to each less file):
````
.global {
  global: global;
}
.import {
  import: import;
}
.button {
  button: button;
}
````

Not sure if related, but the original example has bug: button.less uses `url('../import.less');` while the file name has underscore in it `_import.less`.

I tried multiple import files and all worked relatively too :
* `@import url('./_global.less');`
* `@import url('_global.less');`
* `@import '_global.less';`

So, I think that we need more information.

I used lessc 2.1.1
